runtime/ts/plan/planificator.ts

```diff
@@ -124,22 +134,36 @@ export class Planificator {
     });
   }
 
-  private static async _initStore(arc, {userid, protocol, arcKey}): Promise<StorageProviderBase> {
+  private static async _initSuggestStore(arc, {userid, protocol, arcKey}): Promise<StorageProviderBase> {
     assert(userid, 'Missing user id.');
-    let storage = arc.storageProviderFactory._storageForKey(arc.storageKey);
+    const storage = arc.storageProviderFactory._storageForKey(arc.storageKey);
     const storageKey = storage.parseStringAsKey(arc.storageKey);
     if (protocol) {
       storageKey.protocol = protocol;
     }
     storageKey.location = storageKey.location
         .replace(/\/arcs\/([a-zA-Z0-9_\-]+)$/, `/users/${userid}/suggestions/${arcKey || '$1'}`);
-    const storageKeyStr = storageKey.toString();
-    storage = arc.storageProviderFactory._storageForKey(storageKeyStr);
     const schema = new Schema({names: ['Suggestions'], fields: {current: 'Object'}});
     const type = Type.newEntity(schema);
+    return Planificator._initStore(arc, 'suggestions-id', type, storageKey);
+  }
+
+  private static async _initSearchStore(arc, {userid}): Promise<StorageProviderBase> {
```

Comment:
at some point we will want to add `arc: Arc` type here which will break much of the usages of private APIs below.

Reply:
done. it is better to break sooner rather than later. 

I'm wondering, maybe we could move `location` to KeyBase class, it is present in all volatile, fb, and pouch storage keys, but not in the base.

Reply:
I agree, but can wait for refactoring pr

Reply:
yes, absolutely.